diff --git a/filebrowser/Makefile b/filebrowser/Makefile
index 650e23b1..27b51f30 100644
--- a/filebrowser/Makefile
+++ b/filebrowser/Makefile
@@ -27,7 +27,7 @@ GO_PKG_LDFLAGS_X:= \
        $(GO_PKG)/v2/version.Version=v$(PKG_VERSION)
 
 include $(INCLUDE_DIR)/package.mk
-include ../../lang/golang/golang-package.mk
+include $(TOPDIR)/feeds/packages/lang/golang/golang-package.mk
 
 define Package/filebrowser
   SECTION:=utils
