diff --git a/cdnspeedtest/Makefile b/cdnspeedtest/Makefile
index 80c57d70..1c93afc9 100644
--- a/cdnspeedtest/Makefile
+++ b/cdnspeedtest/Makefile
@@ -24,7 +24,7 @@ GO_PKG:=CloudflareSpeedTest
 GO_PKG_LDFLAGS_X:=main.version=v$(PKG_VERSION)
 
 include $(INCLUDE_DIR)/package.mk
-include ../../lang/golang/golang-package.mk
+include $(TOPDIR)/feeds/packages/lang/golang/golang-package.mk
 
 define Package/cdnspeedtest
   SECTION:=net
