diff --git a/net/pbr/files/etc/config/pbr b/net/pbr/files/etc/config/pbr
index f6532e3..6365889 100644
--- a/net/pbr/files/etc/config/pbr
+++ b/net/pbr/files/etc/config/pbr
@@ -12,7 +12,7 @@ config pbr 'config'
 	option procd_boot_trigger_delay '5000'
 	option procd_reload_delay '0'
 	option resolver_set 'dnsmasq.nftset'
-	option strict_enforcement '1'
+	option strict_enforcement '0'
 	option uplink_interface 'wan'
 	option uplink_interface6 'wan6'
 	option uplink_ip_rules_priority '30000'
