diff --git a/luci-app-lucky/Makefile b/luci-app-lucky/Makefile
index 8ee4d30..c0bb4d7 100644
--- a/luci-app-lucky/Makefile
+++ b/luci-app-lucky/Makefile
@@ -1,7 +1,7 @@
 # SPDX-License-Identifier: GPL-3.0-only
 #
-# Copyright (C) 2022-2023  gdy  
-# 
+# Copyright (C) 2022-2023  gdy
+#
 # This is free software, licensed under the Apache License, Version 2.0 .
 #
 include $(TOPDIR)/rules.mk
@@ -10,18 +10,14 @@ PKG_NAME:=luci-app-lucky
 PKG_VERSION:=2.2.2
 PKG_RELEASE:=1
 
-LUCI_TITLE:=LuCI Support for  lucky 
+LUCI_TITLE:=LuCI Support for lucky
 LUCI_DEPENDS:=+lucky +luci-compat
 LUCI_PKGARCH:=all
 
-
 define Package/$(PKG_NAME)/install
 	$(INSTALL_BIN) ./root/usr/bin/luckyarch $(1)/usr/bin/luckyarch
 endef
 
 include $(TOPDIR)/feeds/luci/luci.mk
 
-
-
-
 # call BuildPackage - OpenWrt buildroot signature
diff --git a/luci-app-lucky/luasrc/controller/lucky.lua b/luci-app-lucky/luasrc/controller/lucky.lua
index c40ceaf..38cc281 100644
--- a/luci-app-lucky/luasrc/controller/lucky.lua
+++ b/luci-app-lucky/luasrc/controller/lucky.lua
@@ -1,147 +1,123 @@
-
 module("luci.controller.lucky", package.seeall)
 
 function index()
-	if not nixio.fs.access("/etc/config/lucky") then
-		return
-	end
-
-	entry({"admin",  "services", "lucky"}, alias("admin", "services", "lucky", "setting"),_("Lucky"), 57).dependent = true
-	entry({"admin", "services", "lucky", "setting"}, cbi("lucky/lucky"), _("Base Setting"), 20).leaf=true
-	entry({"admin", "services", "lucky_status"}, call("act_status"))
-	entry({"admin", "services", "lucky_info"}, call("lucky_info"))
-	entry({"admin", "services", "lucky_set_config"}, call("lucky_set_config"))
-	entry({"admin", "services", "lucky_service"}, call("lucky_service"))
+    if not nixio.fs.access("/etc/config/lucky") then
+        return
+    end
+
+    -- 直接指向设置页面，不再使用alias和二级菜单
+    entry({"admin", "services", "lucky"}, cbi("lucky"), _("Lucky"), 57)
+    entry({"admin", "services", "lucky_status"}, call("act_status"))
+    entry({"admin", "services", "lucky_info"}, call("lucky_info"))
+    entry({"admin", "services", "lucky_set_config"}, call("lucky_set_config"))
+    entry({"admin", "services", "lucky_service"}, call("lucky_service"))
 end
 
-
-
-
 function act_status()
-	local sys  = require "luci.sys"
-	local e = { }
-	e.running = sys.call("pgrep -f 'lucky -c' >/dev/null") == 0
-	luci.http.prepare_content("application/json")
-	luci.http.write_json(e)
+    local sys  = require "luci.sys"
+    local e = { }
+    e.running = sys.call("pgrep -f 'lucky -c' >/dev/null") == 0
+    luci.http.prepare_content("application/json")
+    luci.http.write_json(e)
 end
 
+function lucky_info()
+    local e = { }
+    local luckyInfo = luci.sys.exec("/usr/bin/lucky -info")
 
+    if (luckyInfo~=nil)
+    then
+        e.luckyInfo = luckyInfo
+        local configObj = GetLuckyConfigureObj()
+        if (configObj~=nil)
+        then
+            e.LuckyBaseConfigure = configObj["BaseConfigure"]
+        end
 
+    end
+    e.luckyArch = luci.sys.exec("/usr/bin/luckyarch")
 
-function lucky_info()	
-	local e = { }
-	
-
-	
-	local luckyInfo = luci.sys.exec("/usr/bin/lucky -info")
-	if (luckyInfo~=nil)
-	then
-		e.luckyInfo = luckyInfo
-		local configObj = GetLuckyConfigureObj()
-		if (configObj~=nil)
-		then
-			e.LuckyBaseConfigure = configObj["BaseConfigure"]
-		end
-
-	end
-	e.luckyArch = luci.sys.exec("/usr/bin/luckyarch")
-	--e.runStatus = luci.sys.call("pidof lucky >/dev/null") == 0
-	
-
-	luci.http.prepare_content("application/json")
-	luci.http.write_json(e) 
-end 
-
+    luci.http.prepare_content("application/json")
+    luci.http.write_json(e)
+end
 
 function lucky_set_config()
-	local key = luci.http.formvalue("key") 
-	local value = luci.http.formvalue("value") 
-
-	local e = { }
-	e.ret = 1
-	if (key == "admin_http_port")
-	then
-		e.ret =setLuckyConf("AdminWebListenPort",value)
-	end
-
-	if(key=="admin_safe_url")
-	then
-		e.ret =setLuckyConf("SetSafeURL",value)
-	end
-
-	if(key=="reset_auth_info")
-	then
-		setLuckyConf("AdminAccount","666")
-		setLuckyConf("AdminPassword","666")
-		luci.sys.exec("/etc/init.d/lucky restart")
-		e.ret=0
-	end
-	if(key=="switch_Internetaccess")
-	then
-		e.ret =setLuckyConf("AllowInternetaccess",value)
-	end
-
-
-
-
-	luci.http.prepare_content("application/json")
-	luci.http.write_json(e) 
+    local key = luci.http.formvalue("key")
+    local value = luci.http.formvalue("value")
+    local e = { }
+    e.ret = 1
+    if (key == "admin_http_port")
+    then
+        e.ret = setLuckyConf("AdminWebListenPort", value)
+    end
+
+    if(key == "admin_safe_url")
+    then
+        e.ret = setLuckyConf("SetSafeURL", value)
+    end
+
+    if(key == "reset_auth_info")
+    then
+        setLuckyConf("AdminAccount", "666")
+        setLuckyConf("AdminPassword", "666")
+        luci.sys.exec("/etc/init.d/lucky restart")
+        e.ret = 0
+    end
+    if(key == "switch_Internetaccess")
+    then
+        e.ret = setLuckyConf("AllowInternetaccess", value)
+    end
+
+    luci.http.prepare_content("application/json")
+    luci.http.write_json(e)
 end
 
-
 function lucky_service()
-	local action = luci.http.formvalue("action") 
-	if (action=="restart")
-	then
-		luci.sys.exec("uci set  lucky.@lucky[0].enabled=1")
-		luci.sys.exec("uci commit")
-		luci.sys.exec("/etc/init.d/lucky restart")
-	end
-
-	if (action=="start")
-	then
-		luci.sys.exec("uci set  lucky.@lucky[0].enabled=1")
-		luci.sys.exec("uci commit")
-		luci.sys.exec("/etc/init.d/lucky start")
-	end
-
-	if (action=="stop")
-	then
-		luci.sys.exec("uci set  lucky.@lucky[0].enabled=0")
-		luci.sys.exec("uci commit")
-		luci.sys.exec("/etc/init.d/lucky stop")
-	end
-
+    local action = luci.http.formvalue("action")
+    if (action == "restart")
+    then
+        luci.sys.exec("uci set lucky.@lucky[0].enabled=1")
+        luci.sys.exec("uci commit")
+        luci.sys.exec("/etc/init.d/lucky restart")
+    end
+
+    if (action == "start")
+    then
+        luci.sys.exec("uci set lucky.@lucky[0].enabled=1")
+        luci.sys.exec("uci commit")
+        luci.sys.exec("/etc/init.d/lucky start")
+    end
+
+    if (action == "stop")
+    then
+        luci.sys.exec("uci set lucky.@lucky[0].enabled=0")
+        luci.sys.exec("uci commit")
+        luci.sys.exec("/etc/init.d/lucky stop")
+    end
 end
 
-
 function trim(s)
-	return (string.gsub(s, "^%s*(.-)%s*$", "%1"))
+    return (string.gsub(s, "^%s*(.-)%s*$", "%1"))
 end
 
-
 function GetLuckyConfigureObj()
-	configPath = trim(luci.sys.exec("uci get lucky.@lucky[0].configdir"))
-	local configContent = luci.sys.exec("lucky -baseConfInfo -cd "..configPath)
+    configPath = trim(luci.sys.exec("uci get lucky.@lucky[0].configdir"))
+    local configContent = luci.sys.exec("lucky -baseConfInfo -cd "..configPath)
 
-	configObj = luci.jsonc.parse(trim(configContent))
-	return configObj
+    configObj = luci.jsonc.parse(trim(configContent))
+    return configObj
 end
 
+function setLuckyConf(key, value)
+    configPath = trim(luci.sys.exec("uci get lucky.@lucky[0].configdir"))
 
+    cmd = "/usr/bin/lucky ".." -setconf ".."-key "..key.." -value "..value.." -cd "..configPath
+    if (value == "")
+    then
+        cmd = "/usr/bin/lucky ".." -setconf ".."-key "..key.." -cd "..configPath
+    end
 
-function setLuckyConf(key,value)
-	configPath = trim(luci.sys.exec("uci get lucky.@lucky[0].configdir"))
+    luci.sys.exec(cmd)
 
-	cmd = "/usr/bin/lucky ".." -setconf ".."-key "..key.." -value "..value.." -cd "..configPath
-	if (value=="")
-	then
-		cmd = "/usr/bin/lucky ".." -setconf ".."-key "..key.." -cd "..configPath
-	end
-
-
-	luci.sys.exec(cmd)
-	
-
-	return 0
-end
\ No newline at end of file
+    return 0
+end
diff --git a/luci-app-lucky/luasrc/model/cbi/lucky/lucky.lua b/luci-app-lucky/luasrc/model/cbi/lucky.lua
similarity index 51%
rename from luci-app-lucky/luasrc/model/cbi/lucky/lucky.lua
rename to luci-app-lucky/luasrc/model/cbi/lucky.lua
index 49ee97b..ac52574 100644
--- a/luci-app-lucky/luasrc/model/cbi/lucky/lucky.lua
+++ b/luci-app-lucky/luasrc/model/cbi/lucky.lua
@@ -1,13 +1,14 @@
--- 
-local m, s ,o
+﻿local m, s, o
 
-local m = Map("lucky", translate("Lucky"), translate("ipv4/ipv6 portforward,ddns,reverseproxy proxy,wake on lan,IOT and more...") )
+m = Map("lucky", translate("Lucky"), translate("ipv4/ipv6 portforward, ddns, reverseproxy proxy, wake on lan, IOT and more..."))
 
-m:section(SimpleSection).template  = "lucky/lucky_status"
+-- 状态显示部分
+m:section(SimpleSection).template = "lucky/lucky_status"
 
-s = m:section(TypedSection, "lucky", translate("Basic Settings"))
-s.addremove=false
-s.anonymous=true
+-- 基本设置部分
+s = m:section(TypedSection, "lucky")
+s.anonymous = true
+s.addremove = false
 
 --o=s:option(Flag,"enabled",translate("Enable"))
 --o.default=0
@@ -16,7 +17,6 @@ o = s:option( Value, "configdir", translate("Config dir path"),
 	translate("The path to store the config file"))
 o.placeholder = "/etc/config/lucky.daji"
 
-
 m.apply_on_parse = true
 m.on_after_apply = function(self,map)
 	luci.sys.exec("/etc/init.d/lucky restart")
diff --git a/luci-app-lucky/luasrc/view/lucky/lucky_status.htm b/luci-app-lucky/luasrc/view/lucky/lucky_status.htm
index 098a060..fb17fbb 100644
--- a/luci-app-lucky/luasrc/view/lucky/lucky_status.htm
+++ b/luci-app-lucky/luasrc/view/lucky/lucky_status.htm
@@ -49,7 +49,7 @@
 	  </tr>
 	</table>
   </fieldset>
-  
+
   <fieldset class="cbi-section">
 	<legend><%:Admin Panel Information%></legend>
 	<table>
diff --git a/luci-app-lucky/po/zh_Hans/lucky.po b/luci-app-lucky/po/zh_Hans/lucky.po
index 5762fd5..7822d23 100644
--- a/luci-app-lucky/po/zh_Hans/lucky.po
+++ b/luci-app-lucky/po/zh_Hans/lucky.po
@@ -22,7 +22,7 @@ msgstr "配置文件夹位置"
 msgid "Lucky"
 msgstr "Lucky"
 
-msgid "ipv4/ipv6 portforward,ddns,reverseproxy proxy,wake on lan,IOT and more..."
+msgid "ipv4/ipv6 portforward, ddns, reverseproxy proxy, wake on lan, IOT and more..."
 msgstr "IPv4/IPv6端口转发,动态域名服务,http/https反向代理，网络唤醒(可通过第三方物联网平台接入各大语音助手实现语音控制开关电脑)"
 
 msgid "Running state"
diff --git a/luci-app-lucky/root/usr/share/luci/menu.d/luci-app-lucky.json b/luci-app-lucky/root/usr/share/luci/menu.d/luci-app-lucky.json
deleted file mode 100644
index fbf2b52..0000000
--- a/luci-app-lucky/root/usr/share/luci/menu.d/luci-app-lucky.json
+++ /dev/null
@@ -1,14 +0,0 @@
-{
-    "admin/services/lucky": {
-        "title": "Lucky",
-        "order": 60,
-        "action": {
-            "type": "view",
-            "path": "lucky"
-        },
-        "depends": {
-            "acl": ["luci-app-lucky"],
-            "uci": {"lucky": true}
-        }
-    }
-}
\ No newline at end of file
