diff --git a/tailscale/Makefile b/tailscale/Makefile
index 9039a085..4411a953 100644
--- a/tailscale/Makefile
+++ b/tailscale/Makefile
@@ -33,7 +33,7 @@ GO_PKG_LDFLAGS_X:=tailscale.com/version.shortStamp=$(PKG_VERSION)
 GO_PKG_TAGS:=ts_include_cli,ts_omit_aws,ts_omit_bird,ts_omit_tap,ts_omit_kube,ts_omit_completion
 
 include $(INCLUDE_DIR)/package.mk
-include ../../lang/golang/golang-package.mk
+include $(TOPDIR)/feeds/packages/lang/golang/golang-package.mk
 
 define Package/tailscale
   SECTION:=net
