diff --git a/smartdns/Makefile b/smartdns/Makefile
index 0ea619c..7b7a345 100644
--- a/smartdns/Makefile
+++ b/smartdns/Makefile
@@ -26,7 +26,7 @@ RUST_PKG_FEATURES:=build-release
 RUST_PKG_LOCKED:=0
 
 include $(INCLUDE_DIR)/package.mk
-include ../../lang/rust/rust-package.mk
+include $(TOPDIR)/feeds/packages/lang/rust/rust-package.mk
 
 MAKE_PATH:=src
 MAKE_VARS+= VER=$(PKG_VERSION)
@@ -63,6 +63,10 @@ endef
 
 define Package/smartdns/conffiles
 /etc/config/smartdns
+/etc/smartdns/domain-set
+/etc/smartdns/conf.d
+/etc/smartdns/ip-set
+/etc/smartdns/download
 /etc/smartdns/address.conf
 /etc/smartdns/blacklist-ip.conf
 /etc/smartdns/custom.conf
