diff --git a/luci-app-passwall/luasrc/model/cbi/passwall/client/global.lua b/luci-app-passwall/luasrc/model/cbi/passwall/client/global.lua
index 79c410aa..9c3e7989 100644
--- a/luci-app-passwall/luasrc/model/cbi/passwall/client/global.lua
+++ b/luci-app-passwall/luasrc/model/cbi/passwall/client/global.lua
@@ -152,13 +152,14 @@ o.default = 1070
 o.datatype = "range(1,65535)"
 o.rmempty = false
 o:depends({ node = "", ["!reverse"] = true })
---[[
+
 if has_singbox or has_xray then
 	o = s:taboption("Main", Value, "node_http_port", translate("Node") .. " HTTP " .. translate("Listen Port") .. " " .. translate("0 is not use"))
-	o.default = 0
+	o.default = 10708
 	o.datatype = "port"
+	o:depends({ node = "", ["!reverse"] = true })
 end
-]]--
+
 o = s:taboption("Main", Flag, "node_socks_bind_local", translate("Node") .. " Socks " .. translate("Bind Local"), translate("When selected, it can only be accessed localhost."))
 o.default = "1"
 o:depends({ node = "", ["!reverse"] = true })
