--- a/target/linux/qualcommax/Makefile
+++ b/target/linux/qualcommax/Makefile
@@ -6,3 +6,3 @@ BOARDNAME:=Qualcomm Atheros 802.11ax WiSoC-s
-FEATURES:=squashfs ramdisk fpu nand rtc emmc
+FEATURES:=squashfs ramdisk fpu nand rtc emmc pwm
 KERNELNAME:=Image
--- a/target/linux/qualcommax/ipq807x/config-default
+++ b/target/linux/qualcommax/ipq807x/config-default
@@ -17,2 +17,4 @@ CONFIG_PHYLINK=y
 CONFIG_PINCTRL_IPQ8074=y
+CONFIG_PWM=y
+CONFIG_PWM_IPQ=y
 CONFIG_PINCTRL_QCOM_SPMI_PMIC=y
